#!/usr/bin/env python
PACKAGE="vision_tasks"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("clahe_clip", double_t, 0, "CLAHE clip limit", 4.0, 0.0, 40.0);
gen.add("clahe_grid_size", int_t, 0, "Bilateral filter on CLAHE iterations", 8, 0, 16);
gen.add("clahe_bilateral_iter", int_t, 0, "Bilateral filter on CLAHE iterations", 8, 0, 16);
gen.add("balanced_bilateral_iter", int_t, 0, "Bilateral filter on white balanced iterations", 4, 0, 8);
gen.add("denoise_h", double_t, 0, "Denoising h value", 10.0, 0, 20.0);

gen.add("low_h", int_t, 0, "Lower Bound Of H", 10, 0, 255);
gen.add("high_h", int_t, 0, "Higher Bound Of H", 90, 0, 255);

gen.add("low_s", int_t, 0, "Lower Bound Of S", 0, 0, 255);
gen.add("high_s", int_t, 0, "Higher Bound Of S", 255, 0, 255);

gen.add("low_v", int_t, 0, "Lower Bound Of V", 0, 0, 255);
gen.add("high_v", int_t, 0, "Higher Bound Of V", 255, 0, 255);

gen.add("opening_closing_mat_point", int_t, 0, "", 1, 1, 7);
gen.add("opening_iter", int_t, 0, "Opening iterations", 1, 1, 10);
gen.add("closing_iter", int_t, 0, "Closing iterations", 1, 1, 10);

exit(gen.generate(PACKAGE, "vision_tasks", "buoyRange"));
